mlflow/types/schema.py

```diff
@@ -420,12 +426,22 @@ def _merge(self, obj: "Object") -> "Object":
                 )
 
         """
-        if not isinstance(obj, Object):
-            raise MlflowException(f"Can't merge object with non-object type: {type(obj).__name__}")
-        if self == obj:
+        # Merging object type with AnyType makes all properties optional
```

Comment:
Ah nevermind, I think in the GenAI use-case (e.g. inferring types across a list of chat messages), it's much more useful to still be able to see all the fields rather than just have it be `any`.